tests/commands/create/test_install_app_support_package.py

```diff
@@ -214,22 +298,42 @@ def test_install_pinned_custom_app_support_package_url_with_args(
     myapp.support_package = "https://example.com/custom/support.zip?cool=Yes"
 
     # Write a temporary support zip file
-    support_file = tmp_path / "out.zip"
+    support_file = (
+        tmp_path
+        / "data"
+        / "support"
+        / "55441abbffa311f65622df45a943afc347a21ab40e8dcec79472c92ef467db24"
+        / "out.zip"
+    )
+    support_file.parent.mkdir(parents=True)
     with zipfile.ZipFile(support_file, "w") as support_zip:
         support_zip.writestr("internal/file.txt", data="hello world")
 
     # Modify download_url to return the temp zipfile
     create_command.download_url = mock.MagicMock(return_value=support_file)
 
+    # Mock shutil so we can confirm that unpack is called,
+    # but we still want the side effect of calling it
+    create_command.shutil = mock.MagicMock()
+    create_command.shutil.unpack_archive.side_effect = shutil.unpack_archive
+
     # Install the support package
     create_command.install_app_support_package(myapp)
 
     # Confirm the right URL was used
     create_command.download_url.assert_called_with(
-        download_path=create_command.data_path / "support",
+        download_path=create_command.data_path
+        / "support"
+        / "eb2519992f3776ae59dd0eb1ed6c79fa446a438e2bf4a6d5d61f1adad922f2dd",
         url="https://example.com/custom/support.zip?cool=Yes&revision=42",
     )
 
+    # Confirm the right file was unpacked
+    create_command.shutil.unpack_archive.assert_called_with(
+        support_file,
```

Comment:
Since `support_file` has a different hash to `download_path`, at first sight this doesn't look as if the right file was unpacked, but that's explained by the way that `download_url` is mocked.

 It would be clearer if the mock `download_url` function was made a little more realistic by basing its return value on its `download_path` parameter, rather than a pre-determined value. 

In fact, we could go further by making the mock `download_url` actually create the file as a side-effect. That way, all copies of the ZipFile creation code (lines 308-310) could be factored out into the mock function.

Reply:
Yeah - that's a good idea. I'll make this a proper download side effect function. This is also as good a time as any to add some generic utility functions for download side effects and setting up dummy files.

Reply:
Ok - I've added some utilities to mock file creation and downloads. These functions could be used to clean up a lot of the test codebase, but we can introduce the use of these methods piecemeal (or set them as tasks for contributors who want to get their feet wet with a simple task)